redfish-core/lib/pcie.hpp

```diff
@@ -167,7 +167,10 @@ inline void requestRoutesSystemPCIeDevice(App& app)
                             &pcieDevProperties["DeviceType"]);
                         property)
                     {
-                        asyncResp->res.jsonValue["DeviceType"] = *property;
+                        if (!property->empty())
```

Comment:
From the validator report George shared, we need this check for DeviceClass property in the PcieFunction too.

Reply:
Also the FunctionType property in PcieFunction.

Reply:
when you do this don't forget to update the commit message from:

"Handle empty Device type in PCIeDevice"

to

Handle empty Device type in PCIeDevice and PCIeFunction

since we are broadening the scope of the patch a little bit...

Reply:
@santoshpuranik  Fixed.  
I left them unfixed as I planned of fixing them once we get data related fix for other fields under PCIeFunctions under separate commit but I guess it makes more sense to handle all the empty enum fields under single commit.

Reply:
@bradbishop  Updated.